src/scripts/OSFramework/OSUI/Pattern/AbstractProviderPattern.ts

```diff
@@ -117,6 +121,23 @@ namespace OSFramework.OSUI.Patterns {
 			}
 		}
 
+		/**
+		 * Method that will be responsible to redraw pattern when needed
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.AbstractProviderPattern
+		 */
+		protected redraw(): void {
+			/* If a redraw has been trigger before the previous occured, 
+			cancel the previous one and restart a new redraw process, 
+			this way multiple redraws will be prevented.  */
+			this._cancelRedraw();
+
+			this._refreshTimeOutId = window.setTimeout(() => {
+				this._handleRedraw();
+			}, 0);
```

Comment:
I would prefer us to use the helper `AsyncInvocation`. For that we'd need to change the helper action to return the id of the timer. But It think it would be cleaner or more obvious.

Reply:
well noted! will do it! ;)

Reply:
Done!